Cognite.Jetfire.Api/Credentials.cs

```diff
@@ -22,9 +25,28 @@ public ApiKeyCredentials(string apiKey)
             ApiKey = apiKey;
         }
 
-        public void Apply(HttpRequestMessage request)
+        public async Task ApplyAsync(HttpRequestMessage request)
         {
             request.Headers.Add("api-key", ApiKey);
         }
     }
+
+    public sealed class TokenCredentials : ICredentials
+    {
+        private string Project;
+        private IAuthenticator Authenticator;
+
+        public TokenCredentials(AuthenticatorConfig authenticatorConfig, string project)
+        {
+            Authenticator = new Authenticator(authenticatorConfig, new HttpClient(), NullLogger<IAuthenticator>.Instance);
```

Comment:
This is also something that could be improved in the future. The Authenticator was meant to be constructed using dependency injection, where the http client and the logger are pre-configured with a service provider. Maybe something that can be brought to the jetfire-cli?